sideboard/__init__.py

```diff
@@ -13,6 +14,8 @@
 
 from sideboard.internal.imports import _discover_plugins
 from sideboard.internal.logging import _configure_logging
+import sideboard.run_mainloop
 
-_discover_plugins()
-_configure_logging()
+if 'SIDEBOARD_MODULE_TESTING' not in os.environ:
```

Comment:
This supports the following use case.  Suppose that we want to write a new Python module which bundles a few useful utility functions.  We can do one of two things:
- Write a Python module and put it in ``pypi.python.org`` like most libraries.
- Write a new Sideboard plugin and check it into Github like we do for all our different department-specific plugins.

The first version is sometimes preferable because it allows us to specify the dependency in our ``requirements.txt`` along with all our other deps.  Whereas writing a new Sideboard plugin might require a lot more work, new packages, etc.  (MAGFest probably wouldn't care that much since our puppet setup handles new plugins fairly elegantly.)

The problem came from when the package itself does something like ``from sideboard.lib import DaemonTask``.  Sideboard by default tries to import all of its plugins, which means it checks its own config file and does a ton of stuff that assumes it has a plugins directory and whatnot.  But if we're just e.g. writing unit tests for a library we don't want any of that.  We now allow people to set a ``SIDEBOARD_MODULE_TESTING`` environment variable to turn off plugin loading so that their libraries can just install Sideboard as a library and import it in their unit tests and such.